test/abstractarray.jl

```diff
@@ -1280,3 +1280,26 @@ end
     @test Int[0 t...; t... 0] == [0 1 2; 1 2 0]
     @test_throws ArgumentError Int[t...; 3 4 5]
 end
+
+@testset "hvncat" begin
+    a = fill(1, (2,3,2,4,5))
+    b = fill(2, (1,1,2,4,5))
+    c = fill(3, (1,2,2,4,5))
+    d = fill(4, (1,1,1,4,5))
+    e = fill(5, (1,1,1,4,5))
+    f = fill(6, (1,1,1,4,5))
+    g = fill(7, (2,3,1,4,5))
+    h = fill(8, (3,3,3,1,2))
+    i = fill(9, (3,2,3,3,2))
+    j = fill(10, (3,1,3,3,2))
+
+    result = [a; b c ;;; d e f ; g ;;;;; h ;;;; i j]
+    @test size(result) == (3,3,3,4,7)
+    @test [a; b c ;;; d e f ; g ;;;;; h ;;;; i j] == [a; [b ;; c] ;;; [d e f] ; g ;;;;; h ;;;; i ;; j]
+
+    # terminating semicolons extend dimensions
+    @test [1;] == [1]
+    @test [1;;] == fill(1, (1,1))
+    @test [1,2,3;;] == reshape([1,2,3], (3,1))
+    @test [1,2,3;;;] == reshape([1,2,3], (3,1,1))
+end
```

Comment:
```suggestion
    for v in (1, fill(1), fill(1,1,1), fill(1, 1, 1, 1))
        @test_throws ArgumentError [v; v;; v]
        @test_throws ArgumentError [v; v;; v; v; v]
        @test_throws ArgumentError [v; v; v;; v; v]
        @test_throws ArgumentError [v v; v]
        @test_throws ArgumentError [v v; v v v]
        @test_throws ArgumentError [v v v; v v]
        @test_throws ArgumentError [v; v;; v; v;;; v; v;; v; v;; v; v]
        @test_throws ArgumentError [v; v;; v; v;;; v; v]
        @test_throws ArgumentError [v; v;; v; v;;; v; v; v;; v; v]
        @test_throws ArgumentError [v; v;; v; v;;; v; v;; v; v; v]

        @test [v; v;; v; v] == fill(1, ndims(v) == 3 ? (2, 2, 1) : (2,2))
        @test [v; v;; v; v;;;] == fill(1, 2, 2, 1)
        @test [v; v;; v; v] == fill(1, ndims(v) == 3 ? (2, 2, 1) : (2,2))
        @test [v v; v v;;;] == fill(1, 2, 2, 1)
        @test [v; v;; v; v;;; v; v;; v; v;;] == fill(1, 2, 2, 2)
        @test [v; v; v;; v; v; v;;; v; v; v;; v; v; v;;] == fill(1, 3, 2, 2)
        @test [v v; v v;;; v v; v v] == fill(1, 2, 2, 2)
        @test [v v v; v v v;;; v v v; v v v] == fill(1, 2, 3, 2)
    end
end
```

Reply:
Should I be throwing DimensionMismatch instead? Because that's what `cat` throws for some of those.

Reply:
Of course, hvcat throws ArgumentError.

Reply:
Ah:
```
@test_throws ArgumentError [v v; v]
@test_throws ArgumentError [v v; v v v]
@test_throws ArgumentError [v v v; v v]
```

These go to `hvcat`, which dispatch arrays > 2 dimensions to `cat`, which produces the DimensionMismatch for `fill(1, 1, 1, 1)`. Should those tests be removed?

Reply:
Hah, whoops, I had thought I ran all these tests locally and thought they were fine... I had just figured I'd move my REPL testing into proper tests while I was poking at it.  Hm, yeah, I suppose let's just remove those tests for now since the inconsistency predates you and is unrelated to this work.